src/plugins/discover/public/application/main/data_fetching/fetch_all.ts

```diff
@@ -250,6 +267,18 @@ export async function fetchMoreDocuments(
 
 const isComplete = <T extends DataMsg>(subject: BehaviorSubject<T>) => {
   return subject.pipe(
-    filter(({ fetchStatus }) => [FetchStatus.COMPLETE, FetchStatus.ERROR].includes(fetchStatus))
+    filter(({ fetchStatus }) => [FetchStatus.COMPLETE, FetchStatus.ERROR].includes(fetchStatus)),
+    distinctUntilChanged((a, b) => a.fetchStatus === b.fetchStatus)
```

Comment:
We were actually calling `getDefaultProfileState` twice on document fetch completion since `documents$` gets updated twice and I originally forgot to add `distinctUntilChanged`. Not sure if it actually had an impact, but it could be related to some of the flakiness we've seen lately.